src/EditorFeatures/CSharpTest/CompleteStatement/CSharpCompleteStatementCommandHandlerTests.cs

```diff
@@ -0,0 +1,2577 @@
+    static void Main(string[] args)
+    {
+        string s = ""abcdefghij"";
+        for (int i = s.IndexOf(""bcd""); i < s.IndexOf(""x"");$$ i++)
+";
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ForLoopIncrement()
+        {
+            var code =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        string s = ""abcdefghij"";
+        for (int i = s.IndexOf(""bcd""); i < s.IndexOf(""x""); i = i.IndexOf(""x""$$))
+";
+
+            var expected =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        string s = ""abcdefghij"";
+        for (int i = s.IndexOf(""bcd""); i < s.IndexOf(""x""); i = i.IndexOf(""x"");$$)
+";
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ForLoopBody()
+        {
+            var code =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        string s = ""abcdefghij"";
+        for (int i = s.IndexOf(""bcd""); i < 10; i++)
+        {
+            i.ToString($$)
+        }
+";
+
+            var expected =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        string s = ""abcdefghij"";
+        for (int i = s.IndexOf(""bcd""); i < 10; i++)
+        {
+            i.ToString();$$
+        }
+";
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ForLoopObjectInitializer_MissingParen()
+        {
+            var code =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        for (Goo f = new Goo { i = 0, s = ""abc""$$ }
+    }
+}
+public class Goo
+{
+    public int i;
+    public string s;
+}
+";
+
+            VerifyNoSpecialSemicolonHandling(code);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ForLoopObjectInitializer()
+        {
+            var code =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        for (Goo f = new Goo { i = 0, s = ""abc""$$ } )
+    }
+}
+public class Goo
+{
+    public int i;
+    public string s;
+}
+";
+
+            VerifyNoSpecialSemicolonHandling(code);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ForLoopObjectInitializer_MissingBrace()
+        {
+            var code =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        for (Goo f = new Goo { i = 0, s = ""abc""$$
+    }
+}
+public class Goo
+{
+    public int i;
+    public string s;
+}
+";
+            var expected =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        for (Goo f = new Goo { i = 0, s = ""abc"";$$
+    }
+}
+public class Goo
+{
+    public int i;
+    public string s;
+}
+";
+            VerifyTypingSemicolon(code,expected);
```

Comment:
>VerifyTypingSemicolon [](start = 12, length = 21)

This seems confusing - we have a missing closing brace and closing parent in this test and we don't bail out, but the test above this does have a closing brace and closing paren, but we bail out. I would have expected reverse behavior.

Reply:
Given the object initializer tests below where we bail out for all cases, I think we should bail out for this one as well. Any reason we don't do so?

---
In reply to: [242658763](https://github.com/dotnet/roslyn/pull/30743#discussion_r242658763) [](ancestors = 242658763)

Reply:
Fixed